### ISSUE TITLE
`filename-case` rule false positive

### ISSUE THREAD

**Issue by sindresorhus** [sindresorhus/eslint-plugin-xo#25]
```
spec/iss47-spec.js:9:1
  ✖   9:1   Filename is not in kebab case. Rename it to iss-47-spec.js  xo/filename-case
```

Should we maybe make an exception here? I don't see why we would enforce the number to be independent.

**Comment by jamestalmage** [comment]
It should accept either variant.

**Comment by ntwb** [comment]
I raised a similar case in Gitter https://gitter.im/sindresorhus/ava?at=5726d196e10a59c06107afb2
> `test/configs/i18n.js` - `Filename is not in kebab case. Rename it to i-18-n.js  xo/filename-case`

...files named `i18n.js`, `l10n.js`, or `a11y.js` numeronyms could be ignored, at least in my head I treat these as a single word, and `i-18-n.js` looks pretty weird

**Comment by jfmengels** [comment]
> Nice, looks like #28 takes each of the above into account

Yeah, I saw it pass in the Gitter, so I added it ;)
